salt/_modules/metalk8s.py

```diff
@@ -118,6 +102,39 @@ def add_etcd_node(host, endpoint=None):
         raise exn
 
 
+@depends('KUBERNETES_PRESENT', fallback_function=deps_missing)
+def add_etcd_node(host, endpoint=None):
+    '''Add a new `etcd` node into the `etcd` cluster.
+
+    This module is only runnable from the salt-master on the bootstrap node.
+
+    Arguments:
+        host (str): hostname of the new etcd node
+        endpoint (str): endpoint of the new etcd node
+    '''
+    if not endpoint:
+        # If we have no endpoint get it from mine
+        node_ip = __salt__['saltutil.runner'](
+            'mine.get', tgt=host, fun='control_plane_ip'
+        )[host]
+        endpoint = 'https://{0}:2380'.format(node_ip)
+
+    pod_name = 'etcd-{}'.format(__salt__['network.get_hostname']())
```

Comment:
It's not your code, but I think this will always return the master's hostname (since we run this module from the master). Probably using `saltutil.runner` will work (as was done for `mine.get` just above).

Reply:
Was intended here because we want to connect to the first etcd but yes btw we should not take the hostname of the master but the hostname of the first etcd node

Reply:
Related: #878